README.md

```diff
@@ -53,6 +53,42 @@ You can find a step by step walk through of setting up a backend and accessing i
 
 Additional resources: https://docs.aws.amazon.com/appsync/latest/devguide/building-a-client-app.html 
 
+## Contributing
+
+Contributing guidelines are noted [here](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/blob/master/CONTRIBUTING.md).
+
+### Testing Contributions
+
+If you are contributing to the SDK, it is recommended to add some unit/ functional tests and evaluate against existing tests.
+
+- Unit Tests
+
+  Unit Tests do not require any specific setup and can be run directly from your Xcode IDE.
+
+- Functional Tests
+
+  For running funcitonal tests, we will need the following:
+  - An AppSync API with an `Events App` schema
+  - A Cognito Identity Pool with unauthenticated identities supported
+  - The Cognito Identity Pool's unauth role should have the `AppSync Invoke Full Access` permission
+  
+   You can get the backend setup by following the steps below:
+
+  - Go to [AWS AppSync console](https://console.aws.amazon.com/appsync/home).
+  - Click on `Create New API` and then select `Event App` and hit `Create`, take a note of the `API URL` of the created API.
+  - Once the creation completes, select `Settings` from left side of the console and then select `AWS Identity and Access Management (IAM)` as the authorization type.
+  - Next, create a new Cognito Identity Pool and attach `AppSync Invoke Full Access` permission to the unauth role of the Identity Pool. Keep a note of the Identity Pool ID of newly created pool.
+  - Add a file appsync_test_credentials.json (see sample below) in the `AWSAppSyncTests` folder and replace the values for `AppSyncEndpoint`,  `CognitoIdentityPoolId` and regions if required:
```

Comment:
Pull this out of the bullet list and rephrase it as follows - "Finally, you will need to setup a config file locally to access the server. Add a file ..."